File: swagger/models/contact_log_type.py
from .base_model import BaseModel

from .contact_log_sub_type import ContactLogSubType

class ContactLogType(BaseModel):
	"""
	A contact log type.

	Attributes:

		id
		The contact log type’s ID.

		sub_types
		A list of the subtypes being used to tag this contact log type.
	"""

	input_map = {
		'Id': 'id',
		'SubTypes': ['sub_types', ContactLogSubType],
		}

	output_map = {
		'id': 'Id',
		'sub_types': 'SubTypes',
		}

	id = None
	sub_types = None

File: swagger/Models/ContactLogType.php
<?php

namespace App\Mindbody\Models;

/**
 * A contact log type.
 */
class ContactLogType extends BaseModel
{
	/**
	 * The contact log type’s ID.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * A list of the subtypes being used to tag this contact log type.
	 * 
	 * @var ContactLogSubType[] 
	 */
	public $subTypes;

	protected function getInputMap()
	{
		return [
			'Id' => ['id', 'integer', 'int32'],
			'SubTypes' => ['subTypes', 'array', ContactLogSubType::class],
			];
	}

	protected function getOutputMap()
	{
		return [
			'id' => ['Id', 'integer', 'int32'],
			'subTypes' => ['SubTypes', 'array', ContactLogSubType::class],
			];
	}
}
